scx.full: 1.0.13 -> 1.0.14

Gliczy opened the pull request:
Changelog: https://github.com/sched-ext/scx/releases/tag/v1.0.14
Diff: https://github.com/sched-ext/scx/compare/v1.0.13...v1.0.14

Tests ran successfully and I manually ran all schedulers.
`scx_central` is still non-functional, same as in `1.0.13` (in nixpkgs).

The `substituteInPlace` is needed to compile. It should be removed in the next release, since it already has been changed in master (see: https://github.com/sched-ext/scx/blob/main/lib/scxtest/overrides.h#L16)
<!--
^ Please summarise the changes you have done and explain why they are necessary here ^

For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [Nixpkgs 25.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/doc/release-notes/rl-2511.section.md) (or backporting [25.05](https://github.com/NixOS/nixpkgs/blob/master/doc/manual/release-notes/rl-2505.section.md) Nixpkgs Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
- [NixOS 25.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2511.section.md) (or backporting [25.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) NixOS Release notes)
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md), [pkgs/README.md](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md), [maintainers/README.md](https://github.com/NixOS/nixpkgs/blob/master/maintainers/README.md) and other contributing documentation in corresponding paths.

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#reviewing-contributions
-->

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

Gliczy committed:
scx.full: 1.0.13 -> 1.0.14

JohnRTitor commented:
```
[johnrtitor@build01:~]$ nix build github:nixos/nixpkgs/refs/pull/424862/merge#nixosTests.scx --print-out-paths
warning: killing stray builder process 159619 ()...
warning: killing stray builder process 159696 ()...
warning: killing stray builder process 159697 ()...
warning: killing stray builder process 159699 ()...
warning: killing stray builder process 159700 ()...
warning: killing stray builder process 159703 ()...
error: cycle detected in build of '/nix/store/7k4390h4rf4i5dr0yr6v2jnx0k69kmz3-scx_cscheds-1.0.14.drv' in the references of output 'dev' from output 'out'

```

Nix info on the Nix community builder.

 - system: `"x86_64-linux"`
 - host os: `Linux 6.12.37, NixOS, 25.11 (Xantusia), 25.11.20250711.1bd4d0d`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.28.4`
 - nixpkgs: `/nix/store/9kjmc1d0zgi9b53nk9dif5965nq3app5-source`
 
 There wasn't any error as far as I remember the last time I tried this. I guess something changed on the Nix side.
 
 The only place where it could create a cycle as far I can think of is: https://github.com/NixOS/nixpkgs/blob/9807714d6944a957c2e036f84b0ff8caf9930bc0/pkgs/os-specific/linux/scx/scx_cscheds.nix#L117-L121 where we are building libbpf and the way we are importing these files from the dev output and then using it in rustscheds.
 
 This had to be done this way as the rust binaries wouldn't run properly without properly linking of the compiled libbpf.
 
 I am currently busy with real life and exams, so I can not investigate this right now.

JohnRTitor commented:
Just a simple rebase to master.

I am investigating how to fix this error.

JohnRTitor committed:
scx.cscheds: fix build

by doing this in postFixup, we hopefully don't create a cycle
here obviously this is a dirty fix, but better solutions can be
thought later
Closes https://github.com/NixOS/nixpkgs/issues/425498

Signed-off-by: John Titor <50095635+JohnRTitor@users.noreply.github.com>

JohnRTitor reviewed (approved):
I am merging this as scx is currently broken on master.

We can figure out a more elegant solution later.

JohnRTitor reviewed (approved):
```
[johnrtitor@build01:~]$ nix build github:nixos/nixpkgs/refs/pull/424862/merge#nixosTests.scx --print-out-paths --impure
/nix/store/18dy89rb9znm86xai330r11r40srnb71-vm-test-run-scx_full
```
